opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/EventRestService.java

```diff
@@ -99,6 +99,8 @@ public OnmsEvent getEvent(@PathParam("eventId") final Integer eventId) {
         if (e == null) {
             throw getException(Status.NOT_FOUND, "Event object {} was not found.", Integer.toString(eventId));
         }
+        // Make sure we expose the params sorted:
+        e.setEventParameters(e.getEventParametersInOrder());
```

Comment:
Isn't the `EventDao` the better place for this, as it ensures that ALL calls to load events from the database will have the correct order applied?

Reply:
Ideally we would have one central place where we could ensure the ordering of the event parameters. The ideal place would be OnmsEvent itself which I tried for a while but I couldn't get it to work for various reasons (see https://issues.opennms.org/browse/NMS-9827?focusedCommentId=56051&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-56051).
Other than that I believe there is no other central place where we can ensure the ordering. We can either use
- the DAO level (as you suggested ) which I think would be OnmsEventDaoHibernate, AlarmDaoHibernate, NotificationDaoHibernate, OutageDaoHibernate or
- on the API level
I went for the API level since this seemed less invasive and we order on demand (on those API calls where it is relevant). I fear the API level will be more complicated. But I have no strong opinion on this. I can do either way, whatever is preferred. @mvrueden what do you think?

Reply:
I just don't understand why we need a new `getEventParametersInOrder()` method. 
We could just implement the same logic in `getEventParameters()` or am I missing something?

Reply:
closing this PR in favor of: https://github.com/OpenNMS/opennms/pull/2777